### PR TITLE
Pyre

### LINKED ISSUE #2: Create classes for each sub-process in pyro-processing
Add .h and .cc files for voloxidation, electroreduction, electrorefining, electrowinning.
Combine these classes in a governing pyre.h and .cc file (see Bright lite, Cyder).

### PR DESCRIPTION
**Pull request by gtw2**
Built pyroprocessing model, moving to arfc for group. (not complete - needs more detail and design simulations for ANTPC) Closes #2

### CODE REVIEW

**Comment by gtw2** [comment]
After looking at the files changed, I realized that pride.xml file is out of date - disregard for now I'll update with the proper input format.

**Comment by katyhuff** [comment]
@gtw2 did you mean you would update this pr with a new pride.xml file or we should pull it as is and you'll fix it as a  separate issue later? I haven't seen any activity on this branch since Wednesday.

**Comment by gtw2** [comment]
Yes, I didn't want to clutter the PR with more commits - I'll go ahead and make an issue for it.

File: src/pyre_refining.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+double Refine::Efficiency(double temperature, double pressure, double rotation) {
+  double agitation;
+  double thermal = (8.8333E-7*pow(temperature,3) - 0.00255*(temperature,2)+2.4572*temperature-691.1) / 100;
+  double pres_eff = -0.0055128 * pressure + 100.5;
+  if (rotation <= 1) {
+    agitation = 0.032*rotation + 0.72;
+  } else {
+    agitation = 0.0369924675*log(rotation)+0.829777331;
+    if (agitation > 1) {
+      agitation = 1;
```
Comment (katyhuff) [inline src/pyre_refining.cc]: Shouldn't you throw some kind of error if agitation > 1 ?

File: src/pyre_refining.cc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  Composition::Ptr c = Composition::CreateFromMass(sepcomp);
+  return Material::CreateUntracked(tot_qty, c);
+}
+
+double Refine::Efficiency(double temperature, double pressure, double rotation) {
+  double agitation;
+  double thermal = (8.8333E-7*pow(temperature,3) - 0.00255*(temperature,2)+2.4572*temperature-691.1) / 100;
+  double pres_eff = -0.0055128 * pressure + 100.5;
```
Comment (katyhuff) [inline src/pyre_refining.cc]: Where are these numbers coming from?

File: src/pyre_winning.cc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+using cyclus::toolkit::ResBuf;
+using cyclus::toolkit::MatVec;
+using cyclus::KeyError;
+using cyclus::ValueError;
+using cyclus::Request;
+using cyclus::CompMap;
+
+namespace recycle {
+
+Winning::Winning() {
```
Comment (katyhuff) [inline src/pyre_winning.cc]: Same comment as other default constructors

File: src/pyre_winning.cc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+Material::Ptr Winning::WinningSepMaterial(std::map<int, double> effs, Material::Ptr mat) {
+  CompMap cm = mat->comp()->mass();
+  cyclus::compmath::Normalize(&cm, mat->quantity());
+  double tot_qty = 0;
+  CompMap sepcomp;
+
+  CompMap::iterator it;
+  for (it = cm.begin(); it != cm.end(); ++it) {
+    int nuc = it->first;
+    int elem = (nuc / 10000000) * 10000000;
```
Comment (katyhuff) [inline src/pyre_winning.cc]: big number mult/div thing

File: src/pyre_winning.cc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    sepcomp[nuc] = sepqty;
+    tot_qty += sepqty;
+  }
+
+  Composition::Ptr c = Composition::CreateFromMass(sepcomp);
+  return Material::CreateUntracked(tot_qty, c);
+}
+
+double Winning::Efficiency(double current, double reprocess_time, double flowrate) {
+  double coulombic_eff = -0.00685*pow(current,4) + 0.20413*pow(current,3) - 2.273*pow(current,2) + 11.2046*current - 19.7493;
```
Comment (katyhuff) [inline src/pyre_winning.cc]: line too long. 
Where are these numbers coming from?

Comment (gtw2) [reply src/pyre_winning.cc]: The efficiency numbers came from line fitting literature review data - not the cleanest thing, looking into how to remove it from the code. Perhaps a user-defined file with defaults per my lit review

**Comment by katyhuff** [comment]
first round mostly handled. New issue #5 opened to handle remaining comment.

**Pull request merged**